From 65138c7a4d7a09024041fa5e23eefc1c15f8bd9f Mon Sep 17 00:00:00 2001
From: Sai Pratyusha Magam <smagam@qti.qualcomm.com>
Date: Thu, 11 Dec 2025 18:44:19 +0530
Subject: [PATCH 05/29] 11bi: Add wiphy capability flag for (Re)Association
 frame encryption support

Introduce a new wiphy capability flag to allow AP/STA to
fetch driver support for (Re)Association request/response frame
encryption via the netlink interface NL80211_CMD_GET_WIPHY.

Signed-off-by: Sai Pratyusha Magam <smagam@qti.qualcomm.com>
Signed-off-by: Rohan Dutta <drohan@qti.qualcomm.com>
Signed-off-by: Ainy Kumari <ainy.kumari@oss.qualcomm.com>
---
 src/drivers/driver.h              | 2 ++
 src/drivers/driver_nl80211_capa.c | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/src/drivers/driver.h b/src/drivers/driver.h
index 69f1158f4..172a89aa3 100644
--- a/src/drivers/driver.h
+++ b/src/drivers/driver.h
@@ -2433,6 +2433,8 @@ struct wpa_driver_capa {
 #define WPA_DRIVER_FLAGS2_NON_TRIGGER_BASED_RESPONDER   0x0000000020000000ULL
 /** Driver supports non-trigger based ranging initiator functionality */
 #define WPA_DRIVER_FLAGS2_NON_TRIGGER_BASED_INITIATOR	0x0000000040000000ULL
+/** Driver supports (Re)Association Request/Response frame encryption */
+#define WPA_DRIVER_FLAGS2_ASSOCIATION_FRAME_ENCRYPTION	0x0000000080000000ULL
 
 	u64 flags2;
 
diff --git a/src/drivers/driver_nl80211_capa.c b/src/drivers/driver_nl80211_capa.c
index 2aba1f9a4..5200ddf8d 100644
--- a/src/drivers/driver_nl80211_capa.c
+++ b/src/drivers/driver_nl80211_capa.c
@@ -721,6 +721,10 @@ static void wiphy_info_ext_feature_flags(struct wiphy_info_data *info,
 	if (ext_feature_isset(ext_features, len,
 			      NL80211_EXT_FEATURE_SPP_AMSDU_SUPPORT))
 		capa->flags2 |= WPA_DRIVER_FLAGS2_SPP_AMSDU;
+
+	if (ext_feature_isset(ext_features, len,
+			      NL80211_EXT_FEATURE_ASSOC_FRAME_ENCRYPTION))
+		capa->flags2 |= WPA_DRIVER_FLAGS2_ASSOCIATION_FRAME_ENCRYPTION;
 }
 
 
-- 
2.43.0

